src/helpers.js

```diff
@@ -64,7 +66,30 @@ const autoref = (obj) => {
 };
 
 
+function validateShareMDataPermissions(permissions) {
+  const permissionMustHaveProperties = ['type_tag', 'name', 'perms'];
+  let badPerm = {};
+  const hasCorrectProperties = permissions.every(
+    (perm) => permissionMustHaveProperties.every((prop) => {
+      const bool = Object.prototype.hasOwnProperty.call(perm, prop) && perm[prop];
+      if (!bool) {
+        badPerm = perm;
+        return false;
+      }
+      return true;
+    }));
+
+  if (!hasCorrectProperties) {
+    throw makeFfiError(
```

Comment:
I see an inconsistency here since we are using `makeFfiError` to return an error which contains a `code`, but in other cases we are not returning an error code, even that both validations are made by safe_app_nodejs, e.g. here: https://github.com/maidsafe/safe_app_nodejs/pull/208/files#diff-c9616fc06535a5565aa65b19f33fa08aR181, or  here https://github.com/maidsafe/safe_app_nodejs/pull/208/files#diff-bd9c9dcd314f2d7df52935b3a6a4d504R140.

Perhaps we do want to return error codes from safe_app_nodejs, and we can use the same helper function to do so (we may want to rename `makeFfiError` in that case though), and we'd need to add the list to our documentation.